# Assignment 4 Due Friday, February 24, 2023

This assignment is to be done in teams of 2 people. Please let me know whether:
    
    (i) you have a partner. If so, you and your partner should e-mail me your names
    (ii) you do not yet have a partner. e-mail me and I'll help you find a partner
    
Of course, for this to work, everyone should contact responding to (i) or (ii). Use
the subject line: *Shark Tank team* for both (i) and (ii)

This assignment is an application of Pandas dataframes. In particular, it is to analyze data
about the reality TV show Shark Tank:
    
https://en.wikipedia.org/wiki/Shark_Tank

The task is the following:
    
    You and your partner are an analyst team for Swords Investing (SI).  The investment team
    at SI wants to know whether it should recommend Shark Tank to its clients. You and your
    partner have been assigned the job of developing a prospectus on Shark Tank, summarizing
    and analyzing what happens on the show over the course of the twelve seasons. 
    
    The Shark Tank data is available from Kaggle here in the form of .csv file
        
https://www.kaggle.com/datasets/thirumani/shark-tank-us-dataset
    
Your analysis should include:
        
        * Seasons 1 and 11 and two non-consecutive seasons in between
        * Statistical analysis of the numerical data by season and by show within a season
        * A comparison of the shows within each season
        * An overall comparison between seasons
        * A comparison between those pitches that "Got Deal" and those that didn't
        * An analysis of the categorical data for each season (gender and location)
        * An analysis of the investment amounts of the various panelists
        
**Caution: Pay attention to missing data and how you want to handle it.**

Based on your numerical and categorical anlyses, write a one page prose summary of
your findings.
    
You should use the various pandas commands that are described in Chapter 5 of the text
and in the JupyterLabs for class. Note that the class labs did not cover every item in
the chapter. 

Before diving into the Pandas analysis of the data. Spend some time looking at the data
in the CSV file or in the Pandas dataframe obtained by loading the file. Once you have 
a feel for the data, begin to develop your analysis using Pandas.

You should submit your analysis and summary in a JupyterLab. It should be  clearly 
organized and commented, so that it is easy to follow and understand where different 
analyses are done and summarized.

Your 1 page summmary should appear as the first section of your submission and, of course,
should be written after you've completed your analysis.


## Summary

Across the 4 Seasons we analyzed the following: 
- Total Amount of Money Invested
- Total Number of Deals Made
- Total Amount of Money Invested per Shark (across how many companies)
- Total Proportion of Deals Made by Pitcher's Gender
- Total Proportion of Deals Made by Pitcher's State/Region
    
Across the four seasons analyzed, the sharks invested $\$54,075,665$ across 207 different deals. This means that, on average, $\$261,235.10$ was spent on each deal. But, the sharks did not invest this huge sum of money equally. The sharks that proved to be the most agressive in trying to strike deals were Mark Cuban and Lori Greiner. Barbara Corcoran and Kevin O'Leary were moderate risk takers, while Robert Herjavec and Daymond Johns were the most conserative with their investments. Thus, should one want to game the system and increase their chances of striking a deal with the sharks, they should shoot for around the above average deal cost, and specifically try to win Cuban and Greiner over with the product or idea. As far as the pitcher's  gender goes, male teams/entrepenaurs saw the largest amount of consistent success. Females and Mixed Gender teams/entrepenaurs averaged about the same proportion of deals. Also, from the data available, it appears that the best region to be from is the Pacific Coast or the Northeast. From prior knowledge, this makes sense, based on the proportion of the population of the U.S. located in these regions. 

Seasonal and Episodal analyses are provided below:

## Load and Clean the Data

Once you locate and download the .csv file use the following to load the file: 
(Be careful about your file names. They should have no spaces.)
                                                                

In [1]:
import numpy as np
import pandas as pd
from pandas import Series, DataFrame

In [2]:
Shark_Tank = pd.read_csv('Shark_Tank_US.csv', encoding = "utf-8", encoding_errors = "ignore")

In [3]:
## Drop Unneeded Columns
mylist = ['Season Start', 'Season End', 'Pitch Number', 'Startup Name', 'Industry', 
          'Business Description', 'Pitchers City', 'Pitchers Average Age', 'Entrepreneur Names', 
          'Company Website', 'Multiple Entrepreneurs', 'US Viewership', 'Royalty Deal', 'Loan', 
          'Barbara Corcoran Present', 'Mark Cuban Present', 'Lori Greiner Present', 'Robert Herjavec Present', 
          'Daymond John Present', 'Kevin O Leary Present', 'Kevin Harrington Present', 'Guest Name', 'Notes']
Shark_Tank_lite = Shark_Tank.drop(columns = mylist)

In [4]:
## Create Subsets of DataFrame for Individual Seasons
## Data Frame Indexing Starts at Row 2 in .csv file

## Season 1
Shark_Tank_S1 = Shark_Tank_lite[0:64]

## Season 4
Shark_Tank_S4 = Shark_Tank_lite[160:263]

## Season 7
Shark_Tank_S7 = Shark_Tank_lite[495:611]

## Season 11
Shark_Tank_S11 = Shark_Tank_lite[895:991]

## Analysis

    * Seasons 1 and 11 and two non-consecutive seasons in between
    * Statistical analysis of the numerical data by season and by show within a season
    * A comparison of the shows within each season
    * An overall comparison between seasons
    * A comparison between those pitches that "Got Deal" and those that didn't
    * An analysis of the categorical data for each season (gender and location)
    * An analysis of the investment amounts of the various panelists


Have to find a way to deal with the NaNs --> ex. object.dropna().mean()

### Season 1 Analysis

In [5]:
## Quick Summary of the Data
v_List = ['Episode Number', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity', 'Barbara Corcoran Investment Amount', 
          'Barbara Corcoran Investment Equity', 'Mark Cuban Investment Amount', 'Mark Cuban Investment Equity', 
          'Lori Greiner Investment Amount', 'Lori Greiner Investment Equity', 'Robert Herjavec Investment Amount', 
          'Robert Herjavec Investment Equity', 'Daymond John Investment Amount', 'Daymond John Investment Equity', 
          'Kevin O Leary Investment Amount', 'Kevin O Leary Investment Equity', 'Guest Investment Amount', 
          'Guest Investment Equity']
Shark_Tank_S1[v_List].describe()

Episode Number   Got Deal  Total Deal Amount  Total Deal Equity  \
count       64.000000  64.000000          27.000000          27.000000   
mean         7.281250   0.421875      181111.111111          50.148148   
std          4.130678   0.497763      163984.833570          18.243447   
min          1.000000   0.000000       25000.000000          20.000000   
25%          4.000000   0.000000       50000.000000          45.000000   
50%          7.000000   0.000000      105000.000000          50.000000   
75%         11.000000   1.000000      250000.000000          51.000000   
max         14.000000   1.000000      600000.000000         100.000000   

       Barbara Corcoran Investment Amount  Barbara Corcoran Investment Equity  \
count                           14.000000                           14.000000   
mean                         91607.142857                           33.928571   
std                          75151.175479                           13.035244   
min                          20000.000000                           20.000000   
25%                          50000.000000                           25.000000   
50%                          50000.000000                           27.500000   
75%                         116250.000000                           47.500000   
max                         250000.000000                           55.000000   

       Mark Cuban Investment Amount  Mark Cuban Investment Equity  \
count                           1.0                           1.0   
mean                        50000.0                          20.0   
std                             NaN                           NaN   
min                         50000.0                          20.0   
25%                         50000.0                          20.0   
50%                         50000.0                          20.0   
75%                         50000.0                          20.0   
max                         50000.0                          20.0   

       Lori Greiner Investment Amount  Lori Greiner Investment Equity  \
count                             0.0                             0.0   
mean                              NaN                             NaN   
std                               NaN                             NaN   
min                               NaN                             NaN   
25%                               NaN                             NaN   
50%                               NaN                             NaN   
75%                               NaN                             NaN   
max                               NaN                             NaN   

       Robert Herjavec Investment Amount  Robert Herjavec Investment Equity  \
count                           8.000000                            8.00000   
mean                       113750.000000                           22.18750   
std                        104189.045215                            3.29434   
min                         20000.000000                           17.00000   
25%                         47500.000000                           20.00000   
50%                         75000.000000                           22.50000   
75%                        137500.000000                           25.00000   
max                        300000.000000                           25.50000   

       Daymond John Investment Amount  Daymond John Investment Equity  \
count                        9.000000                        9.000000   
mean                    155277.777778                       32.666667   
std                     144646.359942                       13.973189   
min                      50000.000000                       17.000000   
25%                      52500.000000                       25.000000   
50%                     100000.000000                       25.000000   
75%                     200000.000000                       50.000000   
max                     500000.000

A quick look at the data for Season 1 shows that of the 64 entrepuneours, only 27 got deals. This means that 42.1875\% of the show's contestants were successful. This season, the sharks gave out a total of $\$4,890,000$ dollars, averaging out to about $\$181,111$ dollars per successful deal. Now, how much money each shark invest? Daymond John invested $\$1,397,500$ across 9 companies, Barbara Corcoran invested $\$1,282,500$ dollars across 14 companies, Robert Herjavec invested $\$910,000$ across 8 companies, Kevin O'Leary invested $\$840,000$ across 6 companies, Guests invested $\$410,000$ across 4 companies, Mark Cuban invested $\$50000$ dollars on 1 company and Lori Greiner did not invest in any company.

In [6]:
## Data Grouped by Episode
Shark_Tank_S1[v_List].groupby('Episode Number').sum()

Got Deal  Total Deal Amount  Total Deal Equity  \
Episode Number                                                   
1                      2           510000.0              105.0   
2                      2           750000.0              150.0   
3                      2           285000.0              150.0   
4                      2           235000.0               90.0   
5                      2           260000.0              101.0   
6                      2           200000.0               80.0   
7                      2           350000.0               71.0   
8                      2           190000.0               85.0   
9                      2           300000.0               70.0   
10                     2           625000.0              120.0   
11                     2           550000.0               90.0   
12                     1            40000.0               40.0   
13                     2           305000.0              100.0   
14                     2           290000.0              102.0   

                Barbara Corcoran Investment Amount  \
Episode Number                                       
1                                         280000.0   
2                                          50000.0   
3                                         250000.0   
4                                          35000.0   
5                                         180000.0   
6                                          50000.0   
7                                              0.0   
8                                         140000.0   
9                                         175000.0   
10                                             0.0   
11                                         50000.0   
12                                         20000.0   
13                                         52500.0   
14                                             0.0   

                Barbara Corcoran Investment Equity  \
Episode Number                                       
1                                             80.0   
2                                             20.0   
3                                             50.0   
4                                             40.0   
5                                             50.0   
6                                             50.0   
7                                              0.0   
8                                             55.0   
9                                             45.0   
10                                             0.0   
11                                            40.0   
12                                            20.0   
13                                            25.0   
14                                             0.0   

                Mark Cuban Investment Amount  Mark Cuban Investment Equity  \
Episode Number                                                               
1                                        0.0                           0.0   
2                                    50000.0                          20.0   
3                                        0.0                           0.0   
4                                        0.0                           0.0   
5                                        0.0                           0.0   
6                                        0.0                           0.0   
7                                        0.0                           0.0   
8                                        0.0                           0.0   
9                                        0.0                           0.0   
10                                       0.0                           0.0   
11                                       0.0                           0.0   
12                                       0.0                           0.0   
13                                       0.0                           0.0   
14                                       0.0               

The table above is a brief summary of how each Shark invested their money each epsiode, along with the total amount of money invested per episode. Here we see that, on average, $1.929$ deals made per episode, costing the sharks a whopping $\$349285.71$ per episode. As mentioned before, Barabra Corcoran was the most active shark this season, successfully making a deal in 11 of the 14 episodes. This season, it appears that Mark Cuban and Lori Greiner were mostly along for the ride. 

In [7]:
## Comparison of Entrepenaurs, which ones got a deal which didn't? Gender/Location, etc.
v_List = ['Episode Number', 'Pitchers Gender', 'Pitchers State', 'Original Ask Amount', 'Original Offered Equity', 
          'Valuation Requested', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity']
Shark_Tank_S1[v_List].drop(columns = ['Episode Number']).groupby('Pitchers Gender').sum()

Original Ask Amount  Original Offered Equity  \
Pitchers Gender                                                 
Female                     3201000.0                    430.0   
Male                      10168000.0                    750.0   
Mixed Team                 1720000.0                    208.0   

                 Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers Gender                                                     
Female                    14921666.0        10          1425000.0   
Male                      80646658.0        15          2925000.0   
Mixed Team                10027576.0         2           540000.0   

                 Total Deal Equity  
Pitchers Gender                     
Female                       515.0  
Male                         749.0  
Mixed Team                    90.0

Accounting for the Pitcher's Gender now, of the 27 total deals made, 15 of them were from Pitchers that were Male, 10 from Female, and 2 from Mixed Gender Teams. For those that got deals, Males requested about $\$13557$ per percent equity in their companies, Females were about $\$7444$ per percent equity, and Mixed Teams about $\$8269$ per percent equity. 

In [8]:
Shark_Tank_S1[v_List].drop(columns = ['Episode Number']).groupby('Pitchers State').sum().sort_values('Got Deal', ascending = False)

Original Ask Amount  Original Offered Equity  \
Pitchers State                                                 
CA                        4273000.0                    443.0   
FL                         520000.0                    165.0   
IL                         786000.0                     70.0   
GA                         385000.0                     75.0   
TX                         595000.0                     95.0   
PA                         750000.0                     70.0   
NJ                         660000.0                     40.0   
TN                         255000.0                     50.0   
SC                          50000.0                     25.0   
VA                         350000.0                     20.0   
MD                         150000.0                     30.0   
MA                         200000.0                     25.0   
IN                         125000.0                     25.0   
IA                         180000.0                     20.0   
CO                          55000.0                     25.0   
MN                        1000000.0                     15.0   
NV                         500000.0                     10.0   
NY                        2180000.0                     90.0   
OH                         300000.0                     40.0   
OR                         450000.0                     25.0   
NC                        1325000.0                     30.0   

                Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers State                                                     
CA                       28116242.0         7           860000.0   
FL                        2042158.0         3           195000.0   
IL                        6430000.0         3          1000000.0   
GA                        1795833.0         2           300000.0   
TX                        3500000.0         2           290000.0   
PA                        3500000.0         2           700000.0   
NJ                        5266667.0         1           460000.0   
TN                        1100000.0         1           105000.0   
SC                         200000.0         1            50000.0   
VA                        3500000.0         1            50000.0   
MD                         500000.0         1           200000.0   
MA                         800000.0         1           200000.0   
IN                         500000.0         1           300000.0   
IA                         900000.0         1           180000.0   
CO                         220000.0         0                0.0   
MN                        6666667.0         0                0.0   
NV                        5000000.0         0                0.0   
NY                       18850000.0         0                0.0   
OH                         750000.0         0                0.0   
OR                        3333333.0         0                0.0   
NC                       12625000.0         0                0.0   

                Total Deal Equity  
Pitchers State                     
CA                          335.0  
FL                          172.0  
IL                          180.0  
GA                          105.0  
TX                           90.0  
PA                          100.0  
NJ                           50.0  
TN                           50.0  
SC                           20.0  
VA                           50.0  
MD                           50.0  
MA                           51.0  
IN                           51.0  
IA                           50.0  
CO                            0.0  
MN                            0.0  
NV                            0.0  
NY                            0.0  
OH                            0.0  
OR                            0.0  
NC                            0.0

Now, for the Pitcher's State (grouped by Region). Of the 27 deals made, 7 (26\%) were from the Pacific Coast, 2 (7.4\%) from the Southwest, 0 from the Rocky Mountain Region, 5 (18.5\%) from the Midwest, 8 (29.6\%) from the Southeast, and 5 (18.5\%) from the Northeast. An ANOVA test to compare the mean proportion of deals made by region would likely reject the null hypothesis, suggesting a relationship between the Pitcher's Region and whether they got a deal or not. 

### Season 4 Analysis

In [9]:
## Quick Summary of the Data
v_List = ['Episode Number', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity', 'Barbara Corcoran Investment Amount', 
          'Barbara Corcoran Investment Equity', 'Mark Cuban Investment Amount', 'Mark Cuban Investment Equity', 
          'Lori Greiner Investment Amount', 'Lori Greiner Investment Equity', 'Robert Herjavec Investment Amount', 
          'Robert Herjavec Investment Equity', 'Daymond John Investment Amount', 'Daymond John Investment Equity', 
          'Kevin O Leary Investment Amount', 'Kevin O Leary Investment Equity', 'Guest Investment Amount', 
          'Guest Investment Equity']
Shark_Tank_S4[v_List].describe()

Episode Number    Got Deal  Total Deal Amount  Total Deal Equity  \
count      103.000000  103.000000          52.000000          50.000000   
mean        13.553398    0.504854      182576.923077          31.490000   
std          7.553376    0.502421      162449.430581          15.093075   
min          1.000000    0.000000       25000.000000           8.000000   
25%          7.000000    0.000000       75000.000000          21.250000   
50%         14.000000    1.000000      150000.000000          30.000000   
75%         20.000000    1.000000      250000.000000          35.000000   
max         26.000000    1.000000     1000000.000000         100.000000   

       Barbara Corcoran Investment Amount  Barbara Corcoran Investment Equity  \
count                            9.000000                            9.000000   
mean                        126944.444444                           19.722222   
std                         107541.981467                           14.584048   
min                          25000.000000                            5.000000   
25%                          50000.000000                           10.000000   
50%                          62500.000000                           15.000000   
75%                         150000.000000                           25.000000   
max                         300000.000000                           51.000000   

       Mark Cuban Investment Amount  Mark Cuban Investment Equity  \
count                     18.000000                     18.000000   
mean                  128750.000000                     27.194444   
std                    84035.400454                     21.096514   
min                    50000.000000                      5.000000   
25%                    75000.000000                     15.250000   
50%                   100000.000000                     25.000000   
75%                   150000.000000                     32.250000   
max                   300000.000000                    100.000000   

       Lori Greiner Investment Amount  Lori Greiner Investment Equity  \
count                       13.000000                       13.000000   
mean                    146730.769231                       22.038462   
std                      94568.012427                       14.351918   
min                      37500.000000                        0.000000   
25%                      75000.000000                       15.000000   
50%                     100000.000000                       20.000000   
75%                     200000.000000                       25.000000   
max                     300000.000000                       58.000000   

       Robert Herjavec Investment Amount  Robert Herjavec Investment Equity  \
count                          10.000000                           10.00000   
mean                        99000.000000                           12.65000   
std                         79775.727303                            7.33731   
min                         37500.000000                            0.00000   
25%                         50000.000000                            8.12500   
50%                         75000.000000                           13.75000   
75%                        116250.000000                           16.12500   
max                        300000.000000                           25.00000   

       Daymond John Investment Amount  Daymond John Investment Equity  \
count                       12.000000                       12.000000   
mean                     92625.000000                       26.666667   
std                      89038.838563                       19.315954   
min                      37500.000000                        0.000000   
25%                      46875.000000                       13.125000   
50%                      50000.000000                       27.500000   
75%                      83750.000000                       38.125000   
max                     3

A quick look at the data for Season 4 shows that of the 103 entrepuneours, only 52 got deals. This means that 50.49\% of the show's contestants were successful. This season, the sharks gave out a total of $\$9,494,000$ dollars, averaging out to about $\$182,576$ dollars per successful deal. The sharks gave out roughly double the money as season 1, for about double the amount of deals. Interestly though, the amount of money per deal remains roughly the same. This is starting to suggest that there is a 'sweet spot' of how much money to ask for. Now, how much money each shark invest? Daymond John invested $\$1,111,500$ across 12 companies, Barbara Corcoran invested $\$1,142,500$ dollars across 9 companies, Robert Herjavec invested $\$990,000$ across 10 companies, Kevin O'Leary invested $\$2,025,000$ across 11 companies, Mark Cuban invested $\$2,317,500$ dollars across 18 companies and Lori Greiner invested $\$1,907,500$ across 13 companies. Guests did not invest in any company.

In [10]:
## Data Grouped by Episode
Shark_Tank_S4[v_List].groupby('Episode Number').sum()

Got Deal  Total Deal Amount  Total Deal Equity  \
Episode Number                                                   
1                      2           300000.0               76.0   
2                      2           375000.0               60.0   
3                      2           200000.0               70.0   
4                      2           190000.0               58.0   
5                      2           250000.0               55.0   
6                      2           305000.0               45.0   
7                      2           500000.0               53.0   
8                      2           155000.0               43.0   
9                      2           275000.0               50.0   
10                     1           150000.0               20.0   
11                     2           300000.0               55.0   
12                     2           525000.0               75.0   
13                     3           550000.0               45.0   
14                     1            54000.0               40.0   
15                     2           165000.0               35.0   
16                     2          1100000.0               50.0   
17                     0                0.0                0.0   
18                     2           320000.0              100.0   
19                     2           325000.0               83.0   
20                     3           650000.0              153.0   
21                     2           400000.0               73.0   
22                     3           205000.0               70.5   
23                     2           275000.0               80.0   
24                     2           600000.0               65.0   
25                     2           450000.0               60.0   
26                     3           875000.0               60.0   

                Barbara Corcoran Investment Amount  \
Episode Number                                       
1                                         100000.0   
2                                              0.0   
3                                              0.0   
4                                              0.0   
5                                              0.0   
6                                          55000.0   
7                                              0.0   
8                                              0.0   
9                                          62500.0   
10                                             0.0   
11                                        150000.0   
12                                             0.0   
13                                             0.0   
14                                             0.0   
15                                             0.0   
16                                             0.0   
17                                             0.0   
18                                             0.0   
19                                             0.0   
20                                             0.0   
21                                             0.0   
22                                             0.0   
23                                             0.0   
24                                        450000.0   
25                                             0.0   
26                                        325000.0   

                Barbara Corcoran Investment Equity  \
Episode Number                                       
1                                             56.0   
2                                              0.0   
3                                              0.0   
4                                              0.0   
5                                              0.0   
6                                             15.0   
7                                              0.0   
8                                              0.0   
9                                             10.0   
10                                             0.0   
11    

The table above is a brief summary of how each Shark invested their money each epsiode, along with the total amount of money invested per episode. Here we see that, on average, $2$ deals made per episode, costing the sharks a whopping $\$365153.85$ per episode. This season, Mark Cuban and Lori Greiner stepped up their games from the first season. Cuban was the leading investor, splitting his fortune amonst 18 different companies, while Lori was the second-leading investor with 13 companies. The rest of the sharks averaged about the same number of deals.

In [11]:
## Comparison of Entrepenaurs, which ones got a deal which didn't? Gender/Location, etc.
v_List = ['Episode Number', 'Pitchers Gender', 'Pitchers State', 'Original Ask Amount', 'Original Offered Equity', 
          'Valuation Requested', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity']
Shark_Tank_S4[v_List].drop(columns = ['Episode Number']).groupby('Pitchers Gender').sum()

Original Ask Amount  Original Offered Equity  \
Pitchers Gender                                                 
Female                     4135000.0                    520.0   
Male                      19009000.0                   1164.0   
Mixed Team                 2305000.0                    194.0   

                 Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers Gender                                                     
Female                    28365193.0        13          1710000.0   
Male                     121733726.0        33          6819000.0   
Mixed Team                19510715.0         6           965000.0   

                 Total Deal Equity  
Pitchers Gender                     
Female                       415.0  
Male                         994.5  
Mixed Team                   165.0

Accounting for the Pitcher's Gender now, of the 52 total deals made, 33 of them were from Pitchers that were Male, 13 from Female, and 6 from Mixed Gender Teams. For those that got deals, Males requested about $\$16330$ per percent equity in their companies, Females were about $\$7951.92$ per percent equity, and Mixed Teams about $\$11881$ per percent equity. 

In [12]:
Shark_Tank_S4[v_List].drop(columns = ['Episode Number']).groupby('Pitchers State').sum().sort_values('Got Deal', ascending = False)

Original Ask Amount  Original Offered Equity  \
Pitchers State                                                 
CA                        9650000.0                    519.0   
FL                         920000.0                    159.0   
NJ                         740000.0                     75.0   
WA                         725000.0                     55.0   
CO                         925000.0                    123.0   
TX                        1650000.0                    172.0   
PA                         490000.0                     80.0   
MA                         350000.0                     42.0   
NH                         150000.0                     10.0   
VA                         300000.0                     20.0   
UT                         435000.0                     44.0   
RI                         250000.0                     20.0   
OR                         250000.0                     25.0   
OH                        2600000.0                     32.0   
NY                        1035000.0                    105.0   
NV                         100000.0                     10.0   
AR                          70000.0                     30.0   
AZ                         175000.0                     25.0   
MO                          40000.0                     33.0   
MN                         279000.0                     51.0   
CT                          25000.0                     25.0   
NC                         200000.0                     10.0   
MS                         150000.0                      5.0   
OK                        2000000.0                     10.0   
LA                         600000.0                     30.0   
IL                         150000.0                     20.0   
SC                          50000.0                     40.0   
ID                         150000.0                     10.0   
GA                         140000.0                     30.0   
VT                         600000.0                     35.0   
NE                         250000.0                     33.0   

                Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers State                                                     
CA                       50743913.0        17          2710000.0   
FL                        4269047.0         5           780000.0   
NJ                        6533334.0         5           740000.0   
WA                        6966667.0         3           725000.0   
CO                        7738333.0         3           250000.0   
TX                       10611904.0         2          1450000.0   
PA                        3125000.0         2           300000.0   
MA                        3160714.0         2           325000.0   
NH                        1500000.0         1           300000.0   
VA                        1500000.0         1           300000.0   
UT                        3458333.0         1            50000.0   
RI                        1250000.0         1           250000.0   
OR                        2166667.0         1           100000.0   
OH                       15090909.0         1           600000.0   
NY                       10750000.0         1           250000.0   
NV                        1000000.0         1           100000.0   
AR                         233333.0         1            70000.0   
AZ                        1500000.0         1            75000.0   
MO                         121212.0         1            40000.0   
MN                        1107692.0         1            54000.0   
CT                         100000.0         1            25000.0   
NC                        2000000.0         0                0.0   
MS                        3000000.0         0                0.0   
OK                       20000000.0         0                0.0   
LA                        4000000.0         0                0.0   
IL                         750000.0         0                0.0   

Now, for the Pitcher's State (grouped by Region). Of the 52 deals made, 21 (50\%) were from the Pacific Coast, 3 (5.7\%) from the Southwest, 6 (11.5\%) from the Rocky Mountain Region, 3 (5.7\%) from the Midwest, 7 (13.5\%) from the Southeast, and 12 (23.1\%) from the Northeast. A second ANOVA test would likely have the same conclusion as that for the first season. It appears that the Pacific and Northeast regions have the most consistent likelihood of success. 

### Season 7 Analysis

In [13]:
## Quick Summary of the Data
v_List = ['Episode Number', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity', 'Barbara Corcoran Investment Amount', 
          'Barbara Corcoran Investment Equity', 'Mark Cuban Investment Amount', 'Mark Cuban Investment Equity', 
          'Lori Greiner Investment Amount', 'Lori Greiner Investment Equity', 'Robert Herjavec Investment Amount', 
          'Robert Herjavec Investment Equity', 'Daymond John Investment Amount', 'Daymond John Investment Equity', 
          'Kevin O Leary Investment Amount', 'Kevin O Leary Investment Equity', 'Guest Investment Amount', 
          'Guest Investment Equity']
Shark_Tank_S7[v_List].describe()

Episode Number    Got Deal  Total Deal Amount  Total Deal Equity  \
count      116.000000  116.000000       6.500000e+01          65.000000   
mean        15.000000    0.560345       3.176410e+05          21.841538   
std          8.402898    0.498498       3.812172e+05          10.710799   
min          1.000000    0.000000       5.000000e+04           3.000000   
25%          8.000000    0.000000       1.000000e+05          15.000000   
50%         15.000000    1.000000       2.000000e+05          20.000000   
75%         22.000000    1.000000       3.000000e+05          30.000000   
max         29.000000    1.000000       2.000000e+06          50.000000   

       Barbara Corcoran Investment Amount  Barbara Corcoran Investment Equity  \
count                           13.000000                           13.000000   
mean                        192692.307692                           22.023077   
std                         254366.987053                           13.632630   
min                          50000.000000                            5.000000   
25%                          70000.000000                           10.000000   
50%                         100000.000000                           22.500000   
75%                         200000.000000                           30.000000   
max                        1000000.000000                           50.000000   

       Mark Cuban Investment Amount  Mark Cuban Investment Equity  \
count                  2.200000e+01                     22.000000   
mean                   2.861364e+05                     12.243182   
std                    3.293606e+05                      6.848780   
min                    7.000000e+04                      2.500000   
25%                    1.000000e+05                      8.500000   
50%                    1.500000e+05                     10.000000   
75%                    3.000000e+05                     15.000000   
max                    1.500000e+06                     25.000000   

       Lori Greiner Investment Amount  Lori Greiner Investment Equity  \
count                       20.000000                       20.000000   
mean                    231541.666666                       13.600000   
std                     213010.356357                       10.389266   
min                      25000.000000                        2.500000   
25%                      96250.000000                        9.375000   
50%                     150000.000000                       10.000000   
75%                     262500.000000                       15.375000   
max                     750000.000000                       45.000000   

       Robert Herjavec Investment Amount  Robert Herjavec Investment Equity  \
count                          10.000000                          10.000000   
mean                       199999.933333                          14.080000   
std                        158455.134438                           8.548398   
min                         25000.000000                           5.000000   
25%                        100000.000000                          10.000000   
50%                        133333.000000                          10.000000   
75%                        275000.000000                          14.375000   
max                        500000.000000                          33.300000   

       Daymond John Investment Amount  Daymond John Investment Equity  \
count                        9.000000                        9.000000   
mean                    138333.333333                       22.483333   
std                      90346.554998                       13.152186   
min                      50000.000000                        5.000000   
25%                      75000.000000                       10.000000   
50%                     100000.000000                       25.000000   
75%                     200000.000000                       33.300000   
max                     3

A quick look at the data for Season 7 shows that of the 116 entrepuneours, only 65 got deals. This means that 56.03\% of the show's contestants were successful. This season, the sharks gave out a total of $\$20,646,665$ dollars, averaging out to about $\$317641$ dollars per successful deal. The sharks gave out roughly double the money as season 4, for a slightly greater number of deals. Thus, it makes sense that the amount of money invested per deal roughly doubled from that of season 4. This skews the location of the 'sweet spot' to have a higher likeligood of a deal towards over the $\$200,000$ mark. Now, how much money each shark invest? Daymond John invested $\$1,245,000$ across 9 companies, Barbara Corcoran invested $\$2,505,000$ dollars across 13 companies, Robert Herjavec invested $\$2,000,000$ across 10 companies, Kevin O'Leary invested $\$2,850,593$ across 112 companies, Mark Cuban invested $\$6,295,000$ dollars across 22 companies and Lori Greiner invested $\$4,630,833$ across 20 companies. Guests invested $\$1,120,000$ in 8 companies.

In [14]:
## Data Grouped by Episode
Shark_Tank_S7[v_List].groupby('Episode Number').sum()

Got Deal  Total Deal Amount  Total Deal Equity  \
Episode Number                                                   
1                      2           400000.0               55.0   
2                      2           650000.0               45.0   
3                      2           400000.0               37.5   
4                      2           550000.0               45.0   
5                      2          1600000.0               40.0   
6                      3           600000.0               35.0   
7                      2           325000.0               60.0   
8                      3          1235000.0               70.0   
9                      3           400000.0               66.5   
10                     3          2075000.0               60.0   
11                     3           575000.0               90.0   
12                     2           270000.0               62.0   
13                     2           160000.0               60.0   
14                     3          1050000.0               55.3   
15                     2           500000.0               53.3   
16                     2           400000.0               50.0   
17                     1           210000.0               30.0   
18                     2          1600000.0               35.0   
19                     3           580000.0               65.0   
20                     2           850000.0               37.5   
21                     2           350000.0               37.5   
22                     3          2600000.0               38.0   
23                     3           866666.0               68.3   
24                     2           500000.0               42.0   
25                     1            50000.0               33.3   
26                     1           600000.0                6.7   
27                     3           675000.0               63.5   
28                     2           250000.0               58.3   
29                     2           325000.0               20.0   

                Barbara Corcoran Investment Amount  \
Episode Number                                       
1                                         100000.0   
2                                         300000.0   
3                                              0.0   
4                                              0.0   
5                                              0.0   
6                                              0.0   
7                                         200000.0   
8                                              0.0   
9                                              0.0   
10                                             0.0   
11                                         50000.0   
12                                             0.0   
13                                         60000.0   
14                                             0.0   
15                                             0.0   
16                                             0.0   
17                                         70000.0   
18                                             0.0   
19                                             0.0   
20                                        100000.0   
21                                             0.0   
22                                       1000000.0   
23                                             0.0   
24                                             0.0   
25                                         50000.0   
26                                             0.0   
27                                        475000.0   
28                                             0.0   
29                                        100000.0   

                Barbara Corcoran Investment Equity  \
Episode Number                                       
1                                             15.0   
2                                             12.5   
3                                              0.0   
4                       

The table above is a brief summary of how each Shark invested their money each epsiode, along with the total amount of money invested per episode. Here we see that, on average, $2.24$ deals made per episode, costing the sharks a whopping $\$317,641$ per episode. Again, Mark Cuban and Lori Greiner were the leading investors for this season. Cuban was the leading investor, splitting his fortune amonst 22 different companies, while Lori was the second-leading investor with 20 companies. The rest of the sharks averaged about the same number of deals. Coupled with the increased activity for these two sharks from Season 4, Cuban and Greiner appear to the most 'trigger-happy' sharks, while the rest appear to be more conservative with their investments.

In [15]:
## Comparison of Entrepenaurs, which ones got a deal which didn't? Gender/Location, etc.
v_List = ['Episode Number', 'Pitchers Gender', 'Pitchers State', 'Original Ask Amount', 'Original Offered Equity', 
          'Valuation Requested', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity']
Shark_Tank_S7[v_List].drop(columns = ['Episode Number']).groupby('Pitchers Gender').sum()

Original Ask Amount  Original Offered Equity  \
Pitchers Gender                                                 
Female                     6025000.0                    392.5   
Male                      26520000.0                    866.7   
Mixed Team                 3380000.0                    307.5   

                 Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers Gender                                                     
Female                    67841667.0        14          3745000.0   
Male                     307688606.0        40         15031666.0   
Mixed Team                35420000.0        11          1870000.0   

                 Total Deal Equity  
Pitchers Gender                     
Female                       318.0  
Male                         812.1  
Mixed Team                   289.6

Accounting for the Pitcher's Gender now, of the 65 total deals made, 40 of them were from Pitchers that were Male, 14 from Female, and 10 from Mixed Gender Teams. For those that got deals, Males requested about $\$30598$ per percent equity in their companies, Females were about $\$15350.32$ per percent equity, and Mixed Teams about $\$10645$ per percent equity. This season, homogeneous gendered teams demanded almost double the amount of money per percent equity in their companies as compared to seasons 1 and 4. 

In [16]:
Shark_Tank_S7[v_List].drop(columns = ['Episode Number']).groupby('Pitchers State').sum().sort_values('Got Deal', ascending = False)

Original Ask Amount  Original Offered Equity  \
Pitchers State                                                 
CA                        1100000.0                     40.0   
CO                         250000.0                     26.0   
NY                         850000.0                     82.5   
GA                         100000.0                     10.0   
HI                         200000.0                     10.0   
OH                         200000.0                     10.0   
FL                          50000.0                     20.0   
MA                         550000.0                     23.1   
NV                         750000.0                     40.0   
TX                         400000.0                     10.0   

                Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers State                                                     
CA                       15500000.0         2          1000000.0   
CO                        3583333.0         2           275000.0   
NY                        6791667.0         2           125000.0   
GA                        1000000.0         1           100000.0   
HI                        2000000.0         1           200000.0   
OH                        2000000.0         1           200000.0   
FL                         250000.0         0                0.0   
MA                       16173567.0         0                0.0   
NV                        1875000.0         0                0.0   
TX                        4000000.0         0                0.0   

                Total Deal Equity  
Pitchers State                     
CA                           22.7  
CO                           43.3  
NY                           58.3  
GA                           10.0  
HI                           22.5  
OH                           25.0  
FL                            0.0  
MA                            0.0  
NV                            0.0  
TX                            0.0

Now, for the Pitcher's State (grouped by Region). Unfortunately, for this season, much of this data is missing. But, working with the data for the 9 deals that we have, 3 (33.3\%) were from the Pacific Coast, 0 from the Southwest, 2 (22.2\%) from the Rocky Mountain Region, 1 (11.1\%) from the Midwest, 2 (22.2\%) from the Southeast, and 1 (11.1\%) from the Northeast. Again, since almost 86\% of this data is missing, we will not run any analysis on the small fraction that we do have.

### Season 11 Analysis

In [17]:
## Quick Summary of the Data
v_List = ['Episode Number', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity', 'Barbara Corcoran Investment Amount', 
          'Barbara Corcoran Investment Equity', 'Mark Cuban Investment Amount', 'Mark Cuban Investment Equity', 
          'Lori Greiner Investment Amount', 'Lori Greiner Investment Equity', 'Robert Herjavec Investment Amount', 
          'Robert Herjavec Investment Equity', 'Daymond John Investment Amount', 'Daymond John Investment Equity', 
          'Kevin O Leary Investment Amount', 'Kevin O Leary Investment Equity', 'Guest Investment Amount', 
          'Guest Investment Equity']
Shark_Tank_S11[v_List].describe()

Episode Number   Got Deal  Total Deal Amount  Total Deal Equity  \
count       96.000000  96.000000          63.000000          63.000000   
mean        12.500000   0.656250      302301.587302          19.738889   
std          6.958524   0.477452      246961.903482           9.678007   
min          1.000000   0.000000       35000.000000           3.000000   
25%          6.750000   0.000000      122500.000000          10.000000   
50%         12.500000   1.000000      200000.000000          20.000000   
75%         18.250000   1.000000      400000.000000          25.000000   
max         24.000000   1.000000     1000000.000000          50.000000   

       Barbara Corcoran Investment Amount  Barbara Corcoran Investment Equity  \
count                            7.000000                            7.000000   
mean                        200000.000000                           21.542857   
std                         104083.299973                            9.499799   
min                         100000.000000                           10.000000   
25%                         125000.000000                           13.750000   
50%                         200000.000000                           20.000000   
75%                         225000.000000                           30.000000   
max                         400000.000000                           33.300000   

       Mark Cuban Investment Amount  Mark Cuban Investment Equity  \
count                     17.000000                     17.000000   
mean                  213137.254902                     17.147059   
std                   132545.697509                      7.377679   
min                    33333.333330                      5.000000   
25%                   100000.000000                     11.000000   
50%                   250000.000000                     15.000000   
75%                   300000.000000                     22.000000   
max                   450000.000000                     33.000000   

       Lori Greiner Investment Amount  Lori Greiner Investment Equity  \
count                       21.000000                       21.000000   
mean                    198214.285716                       14.238095   
std                     184242.119859                       10.846219   
min                      17500.000000                        2.500000   
25%                      50000.000000                        8.000000   
50%                     150000.000000                       10.000000   
75%                     250000.000000                       20.000000   
max                     750000.000000                       50.000000   

       Robert Herjavec Investment Amount  Robert Herjavec Investment Equity  \
count                           5.000000                           5.000000   
mean                       273333.333340                          19.000000   
std                        153478.192437                          10.839742   
min                        100000.000000                           5.000000   
25%                        166666.666700                          15.000000   
50%                        300000.000000                          20.000000   
75%                        300000.000000                          20.000000   
max                        500000.000000                          35.000000   

       Daymond John Investment Amount  Daymond John Investment Equity  \
count                        5.000000                        5.000000   
mean                    198500.000000                       23.100000   
std                     181259.482511                        7.536577   
min                      17500.000000                       12.500000   
25%                     125000.000000                       20.000000   
50%                     150000.000000                       25.000000   
75%                     200000.000000                       25.000000   
max                     500000.000

A quick look at the data for Season 11 shows that of the 96 entrepuneours, only 63 got deals. This means that 65.63\% of the show's contestants were successful. This is the greatest success rate of any season analyzed in this report. This season, the sharks gave out a total of $\$19,045,000$ dollars, averaging out to about $\$302301$ dollars per successful deal. The sharks gave out roughly double the same amount of money as season 7, for a slightly greater number of deals. Thus, it makes sense that the amount of money invested per deal is roughly the same as that of season 4. Now, how much money each shark invest? Daymond John invested $\$992,500$ across 5 companies, Barbara Corcoran invested $\$1,400,000$ dollars across 7 companies, Robert Herjavec invested $\$1,366,666$ across 5 companies, Kevin O'Leary invested $\$3,966,666$ across 13 companies, Mark Cuban invested $\$3,623,333$ dollars across 17 companies and Lori Greiner invested $\$4,162,500$ across 21 companies. Guests invested $\$2,933,333$ in 8 companies.

In [18]:
## Data Grouped by Episode
Shark_Tank_S11[v_List].groupby('Episode Number').sum()

Got Deal  Total Deal Amount  Total Deal Equity  \
Episode Number                                                   
1                      3           820000.0              78.00   
2                      2          1750000.0              26.25   
3                      3          1225000.0              54.00   
4                      3           900000.0              35.00   
5                      3           700000.0              38.00   
6                      3           465000.0              50.00   
7                      3           800000.0              72.00   
8                      4          1800000.0              52.50   
9                      2           300000.0              45.00   
10                     2           800000.0              30.00   
11                     3           535000.0              63.00   
12                     3          1050000.0              65.00   
13                     3          1500000.0              63.00   
14                     2           580000.0              30.00   
15                     2           220000.0              58.30   
16                     4           600000.0              76.00   
17                     4          1390000.0              85.00   
18                     2           500000.0              45.00   
19                     1           100000.0              25.00   
20                     2           300000.0              45.00   
21                     2           600000.0              40.00   
22                     2           150000.0              55.00   
23                     1           300000.0              25.00   
24                     4          1660000.0              87.50   

                Barbara Corcoran Investment Amount  \
Episode Number                                       
1                                              0.0   
2                                              0.0   
3                                              0.0   
4                                              0.0   
5                                              0.0   
6                                              0.0   
7                                              0.0   
8                                              0.0   
9                                              0.0   
10                                        400000.0   
11                                             0.0   
12                                        450000.0   
13                                             0.0   
14                                             0.0   
15                                        100000.0   
16                                        150000.0   
17                                             0.0   
18                                             0.0   
19                                             0.0   
20                                        300000.0   
21                                             0.0   
22                                             0.0   
23                                             0.0   
24                                             0.0   

                Barbara Corcoran Investment Equity  \
Episode Number                                       
1                                              0.0   
2                                              0.0   
3                                              0.0   
4                                              0.0   
5                                              0.0   
6                                              0.0   
7                                              0.0   
8                                              0.0   
9                                              0.0   
10                                            20.0   
11                                             0.0   
12                                            42.5   
13                                             0.0   
14                                             0.0   
15                            

The table above is a brief summary of how each Shark invested their money each epsiode, along with the total amount of money invested per episode. Here we see that, on average, $2.625$ deals made per episode, costing the sharks a whopping $\$793,541$ per episode. This is the single largest spend across all seasons of Shark Tank analyzed. Again, Mark Cuban and Lori Greiner were the leading investors for this season. However, this time, Greiner finally out-dealed Cuban. Greiner was the leading investor, splitting her fortune amongst 21 different companies, while Cuban was the second-leading investor with 17 companies. O'Leary and Corcoran were the next group, averaging about 10 deals per season, with Herjavec and John bringing up the rear, being the most conservative sharks.

In [19]:
## Comparison of Entrepenaurs, which ones got a deal which didn't? Gender/Location, etc.
v_List = ['Episode Number', 'Pitchers Gender', 'Pitchers State', 'Original Ask Amount', 'Original Offered Equity', 
          'Valuation Requested', 'Got Deal', 'Total Deal Amount', 'Total Deal Equity']
Shark_Tank_S11[v_List].drop(columns = ['Episode Number']).groupby('Pitchers Gender').sum()

Original Ask Amount  Original Offered Equity  \
Pitchers Gender                                                 
Female                     5975000.0                    354.8   
Male                      14795000.0                    522.5   
Mixed Team                 6860000.0                    248.0   

                 Valuation Requested  Got Deal  Total Deal Amount  \
Pitchers Gender                                                     
Female                    74658633.0        18          4490000.0   
Male                     221477382.0        27          8270000.0   
Mixed Team                92409874.0        18          6285000.0   

                 Total Deal Equity  
Pitchers Gender                     
Female                      399.80  
Male                        482.25  
Mixed Team                  361.50

Accounting for the Pitcher's Gender now, of the 63 total deals made, 27 of them were from Pitchers that were Male, 18 from Female, and 18 from Mixed Gender Teams. For those that got deals, Males requested about $\$28315$ per percent equity in their companies, Females were about $\$19658$ per percent equity, and Mixed Teams about $\$27661$ per percent equity. 

In [20]:
Shark_Tank_S11[v_List].drop(columns = ['Episode Number']).groupby('Pitchers State').sum().sort_values('Got Deal', ascending = False)

Empty DataFrame
Columns: [Original Ask Amount, Original Offered Equity, Valuation Requested, Got Deal, Total Deal Amount, Total Deal Equity]
Index: []

Following the trend from season 7, much of the data we are interested in, we are missing for season 11 as well. Like before, this will not be considered in the final analysis.